extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -335,6 +378,8 @@ public void stop(boolean stopGracefully)
         }
         log.info("Shutdown notice handled");
 
+        taskClient.close();
+        workerExec.shutdownNow();
```

Comment:
@dclim In case of graceful shutdown `workerExec` executor service is used to get the highest offset of tasks in the task group, set that as end offset for all tasks and resume them so that they finish and publish. However, since in most cases doing this will take more time than the time out for shutting down supervisor i.e. `SHUTDOWN_TIMEOUT_MILLIS`(15 seconds), `shutdownNow()` will be called on `workerExec`. Thus, when all the futures for pausing tasks resolves, further logic for getting highest offset and resuming them will not run and fail with exception like -
```
Sep 21, 2016 9:42:33 PM com.google.common.util.concurrent.ExecutionList executeListener
SEVERE: RuntimeException while executing runnable com.google.common.util.concurrent.Futures$ChainingListenableFuture@5fde229f with executor com.google.common.
util.concurrent.MoreExecutors$ListeningDecorator@6c5b645
java.util.concurrent.RejectedExecutionException: Task com.google.common.util.concurrent.Futures$ChainingListenableFuture@5fde229f rejected from java.util.conc
urrent.ThreadPoolExecutor@2fde4963[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 10]
        at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2047)
        at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:823)
        at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1369)
        at com.google.common.util.concurrent.MoreExecutors$ListeningDecorator.execute(MoreExecutors.java:484)
        at com.google.common.util.concurrent.ExecutionList.executeListener(ExecutionList.java:156)
        at com.google.common.util.concurrent.ExecutionList.execute(ExecutionList.java:145)
        at com.google.common.util.concurrent.AbstractFuture.set(AbstractFuture.java:185)
        at com.google.common.util.concurrent.Futures$CombinedFuture.setOneValue(Futures.java:1626)
        at com.google.common.util.concurrent.Futures$CombinedFuture.access$400(Futures.java:1470)
        at com.google.common.util.concurrent.Futures$CombinedFuture$2.run(Futures.java:1548)
        at com.google.common.util.concurrent.MoreExecutors$SameThreadExecutorService.execute(MoreExecutors.java:297)
        at com.google.common.util.concurrent.ExecutionList.executeListener(ExecutionList.java:156)
        at com.google.common.util.concurrent.ExecutionList.execute(ExecutionList.java:145)
        at com.google.common.util.concurrent.ListenableFutureTask.done(ListenableFutureTask.java:91)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:384)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:251)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```